### PR TITLE
add xaasPackager for angular ui

### PR DESCRIPTION
**Pull request by TStadelmaier**
Signed-off-by: Tino Stadelmaier <tstadelmaier.github@gmail.com>

Screenshot
xaaspackager

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

**Comment by koppor** [comment]
Please squash into one commit.

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/CreateFromArtifactApiData.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2017 University of Stuttgart.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v10.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/CreateFromArtifactApiData.java]: Please do not add contributors. EPLv2 does not require to add contributors.

File: org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                               [items]="infrastructureNodetypes"
+                               (data)="refreshSelectedInfrastructureNodeType($event)">
+                    </ng-select>
+                </div>
+            </fieldset>
+        </form>
+    </winery-modal-body>
+
+    <winery-modal-footer [showDefaultButtons]="false">
+        <button type="button" class="btn btn-default" data-dismiss="modal"
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.html]: you can use the defaultButtons here.

it is possible to bind to the click event, as well as whether the button should be disabled or not

File: org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    value: string[];
+
+    @ViewChild('createFromArtifactModal') createFromArtifactModal: ModalDirective;
+
+    constructor(private service: PackagerService,
+                private notify: WineryNotificationService) {
+    }
+
+    ngOnInit() {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.ts]: delete this method if it is not used

File: org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    public refreshValue(value: any): void {
+        this.value = value;
+        this.selectedNodeTypes = this.itemsToStringArray(value);
+    }
+
+    public refreshSelectedArtifactType(value: any) {
+        this.selectedArtifactType = value.text;
+    }
+
+    public refreshSelectedInfrastructureNodeType(value: any) {
```
Comment (lharzenetter) [inline org.eclipse.winery.repository.ui/src/app/section/xaasPackager/xaasPackager.component.ts]: please use the appropriate type here

**Pull request merged**